Add labels and annotations reference stub (#2159)
giantswarm/docs: src/content/reference/k8s-metadata/labels/_index.md

[Source](https://github.com/giantswarm/k8smetadata/blob/v0.24.0/pkg/label/app.go#L9)

### giantswarm.io/aws-ebs-limit

This label is set on EBS CSI driver pods (on AWS) by [aws-ebs-csi-volume-limiter](https://github.com/giantswarm/aws-ebs-csi-volume-limiter/blob/v0.1.0/main.go#L86-L90). It informs about the maximum number of EBS volumes that can be attached to the respective node.

### giantswarm.io/cluster

Used on a variety of resources to associate the resource with a specific cluster. The value is the name of the cluster.

### giantswarm.io/logging

With this label on the Cluster resource, logging via Loki can be disabled. The default value `true` means that logging is enabled. Setting it to `false` disables logging for the entire cluster.

### giantswarm.io/machine-pool

Associates the resource or the node with a node pool, using the name of the node pool as the value.

### giantswarm.io/managed-by

> - TODO Found on various resources in MC and WC. Value differs.
>     - E. g. `rbac-operator` on RBAC resources indicates that rbac-operator created these and will reconcile/delete them.
>     - Value `flux` on an App resource indicates that the App is reconciled by Flux.
>     - Name of a bundle on an App resource indicates that the App has been installed by a bundle.
> 

Reviewer: ```suggestion
Defines which operator is in charge of the resource. Common examples:
    - `rbac-operator` on RBAC resources indicates that the operator created these and will reconcile/delete them.
    - `flux` on an app resource indicates that Flux reconciles the application.
    - A bundle name on an `App` resource indicates that a bundle has installed the App.

```

Author: Thanks!

### giantswarm.io/monitoring

This label set on the Cluster resource with the value `false` can be used to disable the gathering of Prometheus metrics for the entire cluster, and id prevents deployment of Prometheus for this cluster.

On other resources, the label serves as a discovery mechanism for metrics scraping. We're replacing this mechanism by service monitors.

### giantswarm.io/organization

An organization is a key concept in the Giant Swarm platform, used to isolate [tenants]({{< relref "/vintage/platform-overview/multi-tenancy" >}}). This label is used on a variety of resources to associate them with one organization. The value is the name of the organization.

### giantswarm.io/prevent-deletion

Can be set on certain resource types to prevent deletion. See [Prevent accidental deletion of resources]({{< relref "/vintage/advanced/app-platform/deletion-prevention" >}}).

### giantswarm.io/service-type

> - TODO Found on various resources in management and workload clusters.

Reviewer: ```suggestion
```
I would track this out of the docs

- [`fmt`](https://github.com/giantswarm/fmt/blob/278eb4b3318c454e50f24413e7ef2250159f28d6/kubernetes/annotations_and_labels.md?plain=1#L49) has an explanation.
Value is mostly `managed`
- also found occurrence `manual` in an [ops recipe](https://github.com/giantswarm/giantswarm/blob/0d16eb4ebb0440608bb1bfd0636d34afa6352cc6/content/docs/support-and-ops/ops-recipes/cilium-rate-limit-issue.md?plain=1#L39).
- Value `system` is also documented.
- Example use: [PrometheusRule](https://github.com/giantswarm/prometheus-rules/blob/d440f5cc724d9ad2fe4c7f85c9f0b54090a6858e/helm/prometheus-rules/templates/recording-rules/gs-managed-app-deployment-status.rules.yml#L16) - Here it seems to be used to differentiate workloads managed by Giant Swarm from other workloads.

### giantswarm.io/service-priority

Used on cluster-related resources to indicate the [service priority]({{< relref "/vintage/advanced/cluster-management/labelling-workload-clusters#service-priority" >}}), which means the relative importance in general, of the cluster.

### policy.giantswarm.io/psp-status

> - TODO Found on App, Chart, and Cluster resources, MC and WC. On clusters with PSS the value is `disabled`.

Reviewer: @giantswarm/team-shield is still used?

Reviewer 2: It is used internally by a few components related to PSP deprecation/migration. This label can be removed after CAPI migration is complete. But the label itself isn't a customer API

Reviewer: ok then I suggest to remove it

> 
> ### policy.giantswarm.io/psp-status
> 
> - TODO Found on App, Chart, and Cluster resources, MC and WC. On clusters with PSS the value is `disabled`.

Reviewer: ```suggestion
```

### policy.giantswarm.io/resource-kind

- TODO Found in PolicyException and PolicyExceptionDraft resources. Values are like `Deployment`, `CronJob`.

### policy.giantswarm.io/resource-name

- TODO Found in PolicyException and PolicyExceptionDraft resources. Values is the name of the resource the exception is for.

### policy.giantswarm.io/resource-namespace

- TODO Found in PolicyException and PolicyExceptionDraft resources. Values is the namespace of the resource the exception is for.

> 

Reviewer: @giantswarm/team-shield can you provide a good example?

Reviewer: This list is not only about labels that customers can add but is also more of an informational source, even for us. So if we use those labels, let's document them :)

### giantswarm.io/logging

With this label on the Cluster resource, logging via Loki can be disabled. The default value `true` means that logging is enabled. Setting it to `false` disables logging for the entire cluster.

### giantswarm.io/machine-pool

Associates the resource or the node with a node pool, using the name of the node pool as the value.

### giantswarm.io/managed-by

- TODO Found on various resources in MC and WC. Value differs.
    - E. g. `rbac-operator` on RBAC resources indicates that rbac-operator created these and will reconcile/delete them.
    - Value `flux` on an App resource indicates that the App is reconciled by Flux.
    - Name of a bundle on an App resource indicates that the App has been installed by a bundle.

### giantswarm.io/monitoring

This label set on the Cluster resource with the value `false` can be used to disable the gathering of Prometheus metrics for the entire cluster, and id prevents deployment of Prometheus for this cluster.

On other resources, the label serves as a discovery mechanism for metrics scraping. We're replacing this mechanism by service monitors.

> ### giantswarm.io/monitoring_basic_sli

Reviewer 3: This label is getting removed as part of https://github.com/giantswarm/giantswarm/issues/31095

Reviewer 3: Can you remove it already?

Reviewer 3: cc @marians :)

Author: Done
